### PR TITLE
home work #15

### PR DESCRIPTION
**Pull request by DeliriumInTheVoid**

### CODE REVIEW

File: lesson_15/sources/team.cpp [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+{
+    return name_;
+}
+
+void team::add_player(player* player)
+{
+    if (num_players_ < team_max_players_num)
+    {
+        players_[num_players_++] = player;
+        player->team_ = this;
```
Comment (helen-black) [inline lesson_15/sources/team.cpp:17]: it's better to create setter

File: lesson_15/sources/team.cpp [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+{
+    for (int i = 0; i < num_players_; ++i)
+    {
+        if (players_[i] == player)
+        {
+            players_[i]->team_ = nullptr;
+            for (int j = i; j < num_players_ - 1; ++j)
+            {
+                players_[j] = players_[j + 1];
+            }
```
Comment (helen-black) [inline lesson_15/sources/team.cpp:35]: or you can swap with the last player

**Review by helen-black — CHANGES_REQUESTED** [review]

File: lesson_15/sources/team.cpp (hunk 1 above)
Comment (DeliriumInTheVoid) [reply lesson_15/sources/team.cpp:17]: ok.
in this case anyone can set "team" to a "player" but "team" will not know about this change.
and also I done in for using "friend" relationship between classes

File: lesson_15/sources/team.cpp (hunk 2 above)
Comment (DeliriumInTheVoid) [reply lesson_15/sources/team.cpp:35]: ok.
or swap with the first player ) if it ok to change the order of players and there is no guarantees from "team" public API

**Pull request merged**